clang-tidy/action.yaml

```diff
@@ -112,31 +103,35 @@ runs:
     - name: Get target files
       id: get-target-files
       run: |
-        if [ "${{ inputs.target-files }}" != "" ]; then
-          target_files="${{ inputs.target-files }}"
-        else
-          package_path=$(colcon list --paths-only --packages-select ${{ inputs.target-packages }})
-          target_files=$(find $package_path -type f \( -name '*.cpp' -o -name '*.hpp' \) -not -path '*/test/*' -not -path '*/examples/*')
+        read -ra target_packages <<< "${{ inputs.target-packages }}"
+        mapfile -t package_paths < <(colcon list --paths-only --packages-select "${target_packages[@]}")
+        ignore_patterns=()
+        if [ -f .clang-tidy-ignore ]; then
+          while IFS= read -r pattern; do
+            ignore_patterns+=( -not -path "$pattern" )
+          done < .clang-tidy-ignore
         fi
-
+        target_files=$(find "${package_paths[@]}" -type f \( -name '*.cpp' \) "${ignore_patterns[@]}" | tr '\n' ' ')
+        echo "target-files=$target_files"
         echo "target-files=$target_files" >> $GITHUB_OUTPUT
       shell: bash
 
     - name: Analyze
       if: ${{ steps.get-target-files.outputs.target-files != '' }}
       run: |
         mkdir /tmp/clang-tidy-result
-        clang-tidy -p build/ -export-fixes /tmp/clang-tidy-result/fixes.yaml ${{ steps.get-target-files.outputs.target-files }} > /tmp/clang-tidy-result/report.txt || true
+        read -ra target_files <<< "${{ steps.get-target-files.outputs.target-files }}"
+        run-clang-tidy -p build/ -export-fixes /tmp/clang-tidy-result/fixes.yaml -j $(nproc) "${target_files[@]}" > /tmp/clang-tidy-result/report.log || true
         echo "${{ github.event.number }}" > /tmp/clang-tidy-result/pr-id.txt
         echo "${{ github.event.pull_request.head.repo.full_name }}" > /tmp/clang-tidy-result/pr-head-repo.txt
         echo "${{ github.event.pull_request.head.ref }}" > /tmp/clang-tidy-result/pr-head-ref.txt
       shell: bash
 
-    - name: Check if the report.txt file exists
+    - name: Check if the report.log file exists
```

Comment:
Could you write down the motivation and proc/cons of using `.txt` instead of `.log` ?

Reply:
I added description in PR comment.